python/packages/ai/teams/app.py

```diff
@@ -664,115 +664,116 @@ async def on_turn(self, context: TurnContext):
-                    return
-
-            # download input files
-            if (
-                self._options.file_downloaders is not None
-                and len(self._options.file_downloaders) > 0
-            ):
-                input_files = state.temp.input_files if state.temp.input_files is not None else []
-                for file_downloader in self._options.file_downloaders:
-                    files = await file_downloader.download_files(context)
-                    input_files.append(files)
-                state.temp.input_files = input_files
-
-            # run activity handlers
             is_ok, matches = await self._on_activity(context, state)
-
             if not is_ok:
                 await state.save(context, self._options.storage)
                 return
-
-            # only run chain when no activity handlers matched
-            if (
-                matches == 0
-                and self._ai
-                and self._options.ai
-                and context.activity.type == ActivityTypes.message
-                and context.activity.text
-            ):
-                is_ok = await self._ai.run(context, state)
-
-                if not is_ok:
-                    await state.save(context, self._options.storage)
+            if matches == 0:
+                if not await self._run_ai_chain(context, state):
                     return
 
-            # run after turn middleware
-            for after_turn in self._after_turn:
-                is_ok = await after_turn(context, state)
-
-                if not is_ok:
-                    await state.save(context, self._options.storage)
-                    return
+            if not await self._run_after_turn_middleware(context, state):
+                return
 
             await state.save(context, self._options.storage)
         except ApplicationError as err:
             await self._on_error(context, err)
         finally:
             self.typing.stop()
 
+    async def _start_typing(self, context: TurnContext):
+        if self._options.start_typing_timer:
+            await self.typing.start(context)
+
+    def _remove_mentions(self, context: TurnContext):
+        if self.options.remove_recipient_mention and context.activity.type == ActivityTypes.message:
+            context.activity.text = context.remove_recipient_mention(context.activity)
+
+    async def _initialize_state(self, context: TurnContext):
+        state = cast(StateT, await TurnState.load(context, self._options.storage))
+
+        if self._turn_state_factory:
+            state = await self._turn_state_factory(context)
+
+        await state.load(context, self._options.storage)
+        state.temp.input = context.activity.text
+        return state
+
+    async def _authenticate_user(self, context: TurnContext, state):
+        if self.options.auth and self._auth:
+            auth_condition = (
+                isinstance(self.options.auth.auto, bool) and self.options.auth.auto
```

Comment:
do we need to specifically check if `self.options.auth.auto` is True? same for `self.options.auth.auto(context)`